src/wp-includes/formatting.php

```diff
@@ -5390,6 +5390,24 @@ function normalize_whitespace( $str ) {
  * @return string The processed string.
  */
 function wp_strip_all_tags( $string, $remove_breaks = false ) {
+	if ( is_null( $string ) ) {
+		return '';
+	}
+
+	if ( ! is_string( $string ) ) {
+		_doing_it_wrong(
+			__FUNCTION__,
+			sprintf(
+				/* translators: %s: The `$string` argument. */
+				__( 'The %s argument must be a string' ),
+				'$string'
+			),
+			'6.1.0'
+		);
```

Comment:
In PHP < 8 a warning is thrown: `PHP Warning:  strip_tags() expects parameter 1 to be string, array given in /vagrant/wordpress-develop/src/wp-includes/formatting.php on line 5398`

Let's keep the behavour consistent. I can't find a suitable string so I think we'll need a new one

```php
sprintf(
  __( 'Warning: %s expects parameter %d to be a %s, %s given' )
  // etc
)
```

Wrap it all in a trigger error passing `E_USER_WARNING`

Reply:
Updated in [3c654c1](https://github.com/WordPress/wordpress-develop/pull/3132/commits/3c654c1fa2d143c24a181a4614523db09ecabfaa).